tsdb/shard.go

```diff
@@ -1786,154 +1830,245 @@ func (fs *MeasurementFieldSet) IsEmpty() bool {
 type errorChannel chan<- error
 
 type writeRequest struct {
-	done errorChannel
+	errorReturn chan<- error
+	changes     FieldChanges
 }
 
-type MeasurementFieldSetWriter struct {
-	wg            sync.WaitGroup
+type measurementFieldSetChangeMgr struct {
+	mu            sync.Mutex
+	Wg            sync.WaitGroup
```

Comment:
Hopefully this doesn't need to be public?

Reply:
Fixed.  Might have been left over from some call in tsm1/engine.go, I was able to refactor out or something. Not sure.  But fixed.